awx/main/scheduler/kubernetes.py

```diff
@@ -0,0 +1,96 @@
+import time
+
+from django.conf import settings
+from kubernetes.client import Configuration as kubeconfig
+from kubernetes.client import CoreV1Api as v1_kube_api
+from kubernetes.client import ApiClient as kube_api_client
+
+# AWX
+from awx.main.utils.common import parse_yaml_or_json
+from awx.main.utils import (
+    create_temporary_fifo,
+)
+
+
+class PodManager(object):
+    credential = None
+    kube_api = None
+
+    def __init__(self, task):
+        self.task = task
+        self.credential = task.instance_group.credential
+        self.kube_api = self._kube_api_connection(self.credential)
+        self.pod_name = f"job-{task.id}"
+        self.pod_definition = self._pod_definition()
+
+    def deploy_pod(self):
+        if not self.credential.kubernetes:
+            raise RuntimeError
+
+        self.kube_api.create_namespaced_pod(body=self.pod_definition,
+                                            namespace='awx')
+
+        while True:
+            pod = self.kube_api.read_namespaced_pod(name=self.pod_name,
+                                                    namespace=self.namespace)
+            if pod.status.phase != 'Pending':
+                break
+            time.sleep(1)
```

Comment:
We should make a `settings/default.py` for this sleep amount. Maybe a customer can support 500 tower jobs with there super great kubernetes cluster but this could ddos the server.